### ISSUE TITLE
Unclear how to replace flags from `dartdoc`

### ISSUE THREAD

**Issue by akbiggs** [dart-lang/sdk#48313]
`dartdoc` is being replaced with `dart doc`. We have some infra in Fuchsia for generating docs that uses `dartdoc` and uses a few flags that aren't available in `dart doc`. https://cs.opensource.google/fuchsia/fuchsia/+/main:build/dart/gen_reference_docs.py;l=188?q=f:gen_reference&ss=fuchsia

The flags that don't exist and my impressions of how to handle them:

* `--no-validate-links` can be removed. `--validate-links` still exists but `--no-validate-links` doesn't, so I assume the default behavior is not to validate links.
* `--auto-include-dependencies`: I am not sure how to replace this. [Documentation](https://github.com/dart-lang/dartdoc#auto-including-dependencies)
* `--exclude-packages`: We can create a dartdoc_options.yaml file with `exclude` and pass that to dartdoc. That being said I don't know where dartdoc_options.yaml should fit into our build tools (do we predefine it and pass it in?)
* `--format md`: I think this is the default behavior?

Debug info:
* Dart SDK version: 2.17.0-edge.be5198d524368d738e1c810243d4b11db15748c0 (be) (Fri Feb 4 00:30:56 2022 +0000) on "linux_x64"
* Linux

**Comment by akbiggs** [comment]
cc @mit-mit @devoncarew

**Comment by mit-mit** [comment]
cc @srawlins who promised in chat to take a look (but who is out next few days)

**Comment by naudzghebre** [comment]
I'm taking over this issue on the flutter side this week. I was wondering if there is anyone else we could speak to until @srawlins gets back tomorrow?

**Comment by srawlins** [comment]
To unblock you, you should be able to `dart pub global activate dartdoc` and then run the dartdoc binary shipped with the package (`dart run dartdoc`, I think?).

**Comment by naudzghebre** [comment]
We managed to get unblocked by dropping the use of a couple of the flags and disabling dart doc generation for right now.

Ahh I see. I can look into that. But, for clarification, there does not exist of a way currently to pass the flags mentioned above to 'dart doc'?

**Comment by srawlins** [comment]
> But, for clarification, there does not exist of a way currently to pass the flags mentioned above to 'dart doc'?

That's correct. I haven't looked into dartdoc_options.yaml support, but those flags cannot be passed to `dart doc`.

**Comment by srawlins** [comment]
> I don't know where dartdoc_options.yaml should fit into our build tools (do we predefine it and pass it in?)

It should be checked into source control, I would think at as high a position as possible. We don't have a lot of mono-repo customers using this file I think.

> We managed to get unblocked by dropping the use of a couple of the flags

Which flags do you still need? It'd be best if I could add support for them in dartdoc_options.yaml:

* `--auto-include-dependencies` we should add support in dartdoc_options.yaml.
* `--format md`; this is _not_ the default (default is HTML); we should add support in dartdoc_options.yaml.

**Comment by naudzghebre** [comment]
Mainly those two. I think support already exists for `--exclude-packages` in dartdoc_options.yaml if I understand correctly. Unless there is a difference between that flag and `exclude` (which is already supported).
